modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -163,10 +163,10 @@ func (k Keeper) IsActiveChannel(ctx sdk.Context, portID string) bool {
 	return ok
 }
 
-// GetInterchainAccountAddress retrieves the InterchainAccount address from the store keyed by the provided portID
-func (k Keeper) GetInterchainAccountAddress(ctx sdk.Context, portID string) (string, bool) {
+// GetInterchainAccountAddress retrieves the InterchainAccount address from the store associated with the provided connectionID and portID
+func (k Keeper) GetInterchainAccountAddress(ctx sdk.Context, connectionID, portID string) (string, bool) {
 	store := ctx.KVStore(k.storeKey)
-	key := icatypes.KeyOwnerAccount(portID)
+	key := icatypes.KeyOwnerAccount(connectionID, portID)
```

Comment:
technically not necessary for the controller chain since we are guaranteed to have a unique port, but it keeps the code cleaner so I don't think it hurts. Is this the controller or host connection id?

Reply:
Yeah, I figured it was better to keep consistent as both submodules are using the shared genesis types also. 

This is currently using the host connection ID on the controller side, but I would also be interested to hear @AdityaSripal's thoughts. Maybe it makes more sense to use the controller connection ID when setting controller side state?